internal/fields/validate.go

```diff
@@ -820,6 +825,36 @@ func skipValidationForField(key string) bool {
 		isFieldFamilyMatching("event.module", key) // field is deprecated
 }
 
+// skipLeafOfObject checks if the element is a child of an object that was skipped in some previous
+// version of the spec. This is relevant in documents that store fields without subobjects.
+func skipLeafOfObject(root, name string, specVersion semver.Version, schema []FieldDefinition) bool {
+	if specVersion.LessThan(semver3_0_1) {
```

Comment:
Nit. It could be returned early here if spec is spec >= 3.0.1 and change also the latest return as `ancestor != nil`

```golang
func skipLeafOfObject(root, name string, specVersion semver.Version, schema []FieldDefinition) bool {
	if !specVersion.LessThan(semver3_0_1) {
		return false
	}
	// Check if this is a subobject of an object we didn't traverse.
	if !strings.Contains(name, ".") {
		return false
	}
	key := name
	if root != "" {
		key = root + "." + name
	}
	_, ancestor := findAncestorElementDefinition(key, schema, func(key string, def *FieldDefinition) bool {
		// Don't look for ancestors beyond root, these objects have been already traversed.
		if len(key) < len(root) {
			return false
		}
		if !slices.Contains([]string{"group", "object", "nested", "flattened"}, def.Type) {
			return false
		}
		return true
	})

	return ancestor != nil
}

Reply:
Yep, I organized it this way in case we have other cases like this in the future, but happy to change it and refactor later if needed.

Reply:
Refactored, thanks.